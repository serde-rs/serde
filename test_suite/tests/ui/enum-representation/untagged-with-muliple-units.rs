use serde_derive::{Deserialize, Serialize};

#[derive(Deserialize)]
#[serde(untagged)]
enum E1 {
    Unit1,
    Tuple1(usize),
    Unit2,
    Struct {},
    #[serde(skip_serializing)]
    Unit3,
    #[serde(skip_deserializing)]
    Unit4,
}

#[derive(Serialize)]
#[serde(untagged)]
enum E2 {
    Unit1,
    Tuple1(usize),
    Unit2,
    Struct {},
    #[serde(skip_serializing)]
    Unit3,
    #[serde(skip_deserializing)]
    Unit4,
}

fn main() {}
