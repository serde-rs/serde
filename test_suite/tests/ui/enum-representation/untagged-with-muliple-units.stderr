error: untagged enums can contain only one unit variant. Use #[serde(skip_deserializing)] if you really want several unit variants
 --> tests/ui/enum-representation/untagged-with-muliple-units.rs:8:5
  |
8 |     Unit2,
  |     ^^^^^

error: untagged enums can contain only one unit variant. Use #[serde(skip_deserializing)] if you really want several unit variants
  --> tests/ui/enum-representation/untagged-with-muliple-units.rs:10:5
   |
10 | /     #[serde(skip_serializing)]
11 | |     Unit3,
   | |_________^
